src/main/java/io/gatling/mojo/Interactive.java

```diff
@@ -0,0 +1,70 @@
+
+/*
+ * Copyright 2011-2022 GatlingCorp (https://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gatling.mojo;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.Scanner;
+import org.apache.maven.plugin.MojoFailureException;
+
+public class Interactive {
+
+  private static final int MAX_INTERACTIVE_SIMULATION_SELECT_ATTEMPTS = 5;
+
+  static String selectSingleSimulation(List<String> simulations) throws MojoFailureException {
+    return selectSingleSimulationRec(new Scanner(System.in), simulations, 0);
+  }
+
+  private static String selectSingleSimulationRec(
+      Scanner scanner, List<String> simulations, int attempts) throws MojoFailureException {
+    if (attempts > MAX_INTERACTIVE_SIMULATION_SELECT_ATTEMPTS) {
+      throw new MojoFailureException(
+          "Max attempts of reading simulation number ("
+              + MAX_INTERACTIVE_SIMULATION_SELECT_ATTEMPTS
+              + ") reached. Aborting.");
+    } else {
```

Comment:
nit: I like to omit else clause when throwing or doing an early return, to reduce nesting a bit and highlight preconditions (it's a matter of personal taste)

Reply:
good point